### DISCUSSION TITLE
How to acesss to Slider Track?

### DISCUSSION THREAD

**Discussion by sn4k3** [AvaloniaUI/Avalonia#6075] [Q&A] [answered]
Currently i had replicated the Slider.cs code just to expose Track with a public getter. But this seens ugly and unnecessary...
Is there any way on code i can get the Track component of the Slider?

Var: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Slider.cs#L84

**Comment by maxkatz6** [comment, marked as answer]
In TemplateApplied event/override you can get it from the namescope in similar way as it is here:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Slider.cs#L189

Other way, less efficient, is to find this control in the visual tree (traverse VisualChildren enumerable recursively).

**Reply by sn4k3** [reply]
Yes that worked!
I tried `var track = LayerSlider.FindNameScope().Find<Track>("PART_Track");` before on OnLoad but without success

But inside the event it worked fine:

```C#
LayerSlider.TemplateApplied += (sender, e) =>
{
     var track = e.NameScope.Find<Track>("PART_Track");
};
```

Thanks
